core/properties/WeylTensor.cc

```diff
@@ -22,10 +22,21 @@ std::string WeylTensor::name() const
 	return "WeylTensor";
 	}
 
+// Traceless and TableauSymetry args are overwritten elsewhere
+bool WeylTensor::parse(Kernel&, keyval_t& keyvals)
+	{
+	return true;
+	}
+
 void WeylTensor::validate(const Kernel& kernel, const Ex& pat) const
 	{
 	if(Algorithm::number_of_indices(kernel.properties, pat.begin())!=4)
 		throw ConsistencyException("WeylTensor: need exactly 4 indices.");
+	index_iterator indit=index_iterator::begin(kernel.properties, pat.begin());
+	auto ind=kernel.properties.get<Indices>(indit, true);
+	// We cannot access the right things from parse()
+	WeylTensor *ptr = const_cast<WeylTensor*>(this);
+	ptr->index_set_name=ind->set_name;
```

Comment:
Thanks for pointing this out. At the moment leave this hack in, but it is clear that we need a better way to do this, probably by passing an iterator into `parse`  which points to the object to which the object is being attached, or by making `validate` non-const.